app/src/main/java/com/bunbeauty/papakarlo/feature/address/screen/createaddress/CreateAddressFragment.kt

```diff
@@ -112,115 +108,92 @@ class CreateAddressFragment : BaseFragmentWithSharedViewModel(R.layout.layout_co
                         .padding(16.dp)
                 ) {
                     val focusManager = LocalFocusManager.current
-                    var streetText by remember {
-                        mutableStateOf("")
-                    }
-                    var expanded by remember(state.suggestedStreetList) {
-                        mutableStateOf(state.suggestedStreetList.isNotEmpty())
+                    var expanded by remember(viewState.suggestionListNotEmpty) {
+                        mutableStateOf(viewState.suggestionListNotEmpty)
                     }
+
                     FoodDeliveryTextFieldWithMenu(
                         modifier = Modifier
                             .fillMaxWidth()
                             .onFocusChanged { focusState ->
-                                expanded =
-                                    focusState.isFocused && state.suggestedStreetList.isNotEmpty()
+                                onAction(CreateAddress.Action.StreetFocusChange(isFocused = focusState.isFocused))
+                                expanded = focusState.isFocused && viewState.suggestionListNotEmpty
                             },
                         expanded = expanded,
                         onExpandedChange = { value ->
                             expanded = value
                         },
-                        value = streetText,
+                        onSuggestionClick = { suggestion ->
+                            focusManager.moveFocus(FocusDirection.Down)
+                            onAction(CreateAddress.Action.SuggestionSelect(suggestion = suggestion))
+                        },
+                        value = viewState.street,
                         labelStringId = R.string.hint_create_address_street,
-                        onValueChange = { value ->
-                            streetText = value
-                            viewModel.onStreetTextChanged(value)
+                        onValueChange = { street ->
+                            onAction(CreateAddress.Action.StreetTextChange(street = street))
                         },
-                        errorMessageId = if (state.hasStreetError) {
+                        errorMessageId = if (viewState.hasStreetError) {
```

Comment:
you could create viewState and map to streetErrorMessage : Int?
and same for other fields

Reply:
✅ fixed